pkg/scheduler/scheduler_test.go

```diff
@@ -643,3 +643,372 @@ func indexByPodAnnotationNodeName(obj interface{}) ([]string, error) {
+					{PluginName: queueSortPlugin, QueueingHintFn: defaultQueueingHintFn},
+				},
+				{Resource: framework.PersistentVolume, ActionType: framework.All}: {
+					{PluginName: bindPlugin, QueueingHintFn: defaultQueueingHintFn},
+					{PluginName: queueSortPlugin, QueueingHintFn: defaultQueueingHintFn},
+				},
+				{Resource: framework.StorageClass, ActionType: framework.All}: {
+					{PluginName: bindPlugin, QueueingHintFn: defaultQueueingHintFn},
+					{PluginName: queueSortPlugin, QueueingHintFn: defaultQueueingHintFn},
+				},
+				{Resource: framework.PersistentVolumeClaim, ActionType: framework.All}: {
+					{PluginName: bindPlugin, QueueingHintFn: defaultQueueingHintFn},
+					{PluginName: queueSortPlugin, QueueingHintFn: defaultQueueingHintFn},
+				},
+				{Resource: framework.PodSchedulingContext, ActionType: framework.All}: {
+					{PluginName: bindPlugin, QueueingHintFn: defaultQueueingHintFn},
+					{PluginName: queueSortPlugin, QueueingHintFn: defaultQueueingHintFn},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			registry := frameworkruntime.Registry{}
+			cfgPls := &schedulerapi.Plugins{}
+			for _, pl := range tt.plugins {
+				tmpPl := pl
+				if err := registry.Register(pl.Name(), func(_ runtime.Object, _ framework.Handle) (framework.Plugin, error) {
+					return tmpPl, nil
+				}); err != nil {
+					t.Fatalf("fail to register filter plugin (%s)", pl.Name())
+				}
+				cfgPls.Filter.Enabled = append(cfgPls.Filter.Enabled, schedulerapi.Plugin{Name: pl.Name()})
+			}
+
+			profile := schedulerapi.KubeSchedulerProfile{Plugins: cfgPls}
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			fwk, err := newFrameworkWithQueueSortAndBind(registry, profile, stopCh)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			exts := fwk.EnqueueExtensions()
+			// need to sort to make the test result stable.
+			sort.Slice(exts, func(i, j int) bool {
+				return exts[i].Name() < exts[j].Name()
+			})
+
+			got := buildQueueingHintMap(exts)
+
+			for e, fns := range got {
+				wantfns, ok := tt.want[e]
+				if !ok {
+					t.Errorf("got unexpected event %v", e)
+					continue
+				}
+				if len(fns) != len(wantfns) {
+					t.Errorf("got %v queueing hint functions, want %v", len(fns), len(wantfns))
+					continue
+				}
+				for i, fn := range fns {
+					if fn.PluginName != wantfns[i].PluginName {
+						t.Errorf("got plugin name %v, want %v", fn.PluginName, wantfns[i].PluginName)
+						continue
+					}
+					if fn.QueueingHintFn(nil, nil, nil) != wantfns[i].QueueingHintFn(nil, nil, nil) {
+						t.Errorf("got queueing hint function (%v) returning %v, expect it to return %v", fn.PluginName, fn.QueueingHintFn(nil, nil, nil), wantfns[i].QueueingHintFn(nil, nil, nil))
+						continue
+					}
+				}
+			}
+		})
+	}
+}
+
+// Test_UnionedGVKs tests UnionedGVKs worked with buildQueueingHintMap.
+func Test_UnionedGVKs(t *testing.T) {
```

Comment:
This test proves currently even if we register any plugins, we register the same in event handlers. (It's existing behavior, not due to the change of this PR.)

It's because the default queue sort plugin etc don't implement EventsToRegister, but we apply default EventsToRegister on such plugins in NewFramework().
Probably, we don't need to do that when it comes to queuesort plugin or other re-enqueue unrelated plugins if they don't have EventsToRegister.
I'll follow up in another PR after this PR.